src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+const liElements = document.querySelectorAll('li');
+
+liElements.forEach((li) => {
+  const ul = li.querySelector('ul');
+
+  if (ul) {
+    const span = document.createElement('span');
+    const first = li.firstChild;
```

Comment:
The variable 'first' assumes that the first child of 'li' is a text node. If there are other types of nodes (like comments or elements) before the text node, this could lead to unexpected behavior. Consider checking if 'first' is indeed a text node before using 'textContent'.